common/cpp/src/google_smart_card_common/global_context.h

```diff
@@ -0,0 +1,42 @@
+// Copyright 2020 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_SMART_CARD_COMMON_GLOBAL_CONTEXT_H_
+#define GOOGLE_SMART_CARD_COMMON_GLOBAL_CONTEXT_H_
+
+#include <google_smart_card_common/value.h>
+
+namespace google_smart_card {
+
+// Global context is an interface that abstracts away webport-specific
+// operations.
+//
+// The implementation of this class is required to be thread-safe.
+class GlobalContext {
```

Comment:
WDYT? It's not the most elegant interface, admittedly, but it's somewhat dictated by the current code structure. For example, it's a drop-in replacement for the `JsRequester::Delegate` delegate: https://github.com/GoogleChromeLabs/chromeos_smart_card_connector/blob/6257dbf03a65b721b007d5bd1e044ca23163fd75/common/cpp/src/google_smart_card_common/requesting/js_requester.h#L51

As an alternative, for example, I'd rather see this as an `MessagePipe` interface with methods like `PostMessages()` and `RegisterTypedMessageHandler()`. But doing that now would increase the already big amount of refactoring to be done.

Reply:
I think this is fine for now. We can tackle any other refactoring after successfully migrating to Wasm.